bug(Duolingo - Unlock super): App crashes with a java verify error

### Type

Error at runtime

### Bug description

In new release there is a patch to duolingo to unlock super. https://github.com/ReVanced/revanced-patches/pull/2862
After patching duolingo in app and installing it I can't login to app.
After clicking `Sign in` button is changing state to loading and nothing more happen.


### Steps to reproduce

Step 1: Download duolingo apk. Tested with v 5.116.4 and 5.117.4. Source: apkmirror
Step 2: Patch apk in revanced app
Step 3: Install patched apk
Step 4: open app and click `I already have an account` and fill form
Step 5: click `sign in` and wait endlessly

### Relevant log output

```shell
FATAL EXCEPTION: RxComputationThreadPool-1
08-28 12:32:03.020 10709 10828 E AndroidRuntime: Process: com.duolingo, PID: 10709
08-28 12:32:03.020 10709 10828 E AndroidRuntime: wk.e: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.lang.VerifyError: Verifier rejected class com.duolingo.user.q: void com.duolingo.user.q.<init>(com.duolingo.ads.AdsConfig, c4.k, com.duolingo.user.BetaStatus, java.lang.String, org.pcollections.l, org.pcollections.l, boolean, com.duolingo.shop.Outfit, org.pcollections.l, long, c4.m, com.duolingo.core.legacymodel.Direction, java.lang.String, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, org.pcollections.h, java.lang.String, org.pcollections.h, com.duolingo.shop.h, com.duolingo.user.GlobalAmbassadorStatus, java.lang.String, boolean, boolean, boolean, boolean, y7.f, java.lang.String, org.pcollections.l, long, int, java.lang.String, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.l, org.pcollections.l, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.h, org.pcollections.l, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, ia.j, boolean, org.pcollections.l, org.pcollections.l, org.pcollections.h, boolean, boolean, boolean, com.duolingo.user.StreakData, org.pcollections.l, java.lang.String, long, y4.r, java.lang.String, org.pcollections.l, com.duolingo.session.hi, boolean, com.duolingo.user.o, boolean, boolean, org.pcollections.l) failed to verify: void com.duolingo.user.q.<init>(com.duolingo.ads.AdsConfig, c4.k, com.duolingo.user.BetaStatus, java.lang.String, org.pcollections.l, org.pcollections.l, boolean, com.duolingo.shop.Outfit, org.pcollections.l, long, c4.m, com.duolingo.core.legacymodel.Direction, java.lang.String, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, org.pcollections.h, java.lang.String, org.pcollections.h, com.duolingo.shop.h, com.duolingo.user.GlobalAmbassadorStatus, java.lang.String, boolean, boolean, boolean, boolean, y7.f, java.lang.String, org.pcollections.l, long, int, java.lang.String, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.l, org.pcollections.l, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.h, org.pcollections.l, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, ia.j, boolean, org.pcollections.l, org.pcollections.l, org.pcollections.h, boolean, boolean, boolean, com.duolingo.user.StreakData, org.pcollections.l, java.lang.String, long, y4.r, java.lang.String, org.pcollections.l, com.duolingo.session.hi, boolean, com.duolingo.user.o, boolean, boolean, org.pcollections.l): [0x14] register v11 has type Undefined but expected Precise Reference: java.lang.String (declaration of 'com.duolingo.user.q' appears in /data/app/~~bvl59fr-yJPGNvh67taehQ==/com.duolingo-bfnCO2Wma8kKJsNNeVKGpQ==/base.apk!classes4.dex)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at ql.a.b(Unknown Source:45)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at il.l.run(Unknown Source:62)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at il.l.call(Unknown Source:0)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1012)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: Caused by: java.lang.VerifyError: Verifier rejected class com.duolingo.user.q: void com.duolingo.user.q.<init>(com.duolingo.ads.AdsConfig, c4.k, com.duolingo.user.BetaStatus, java.lang.String, org.pcollections.l, org.pcollections.l, boolean, com.duolingo.shop.Outfit, org.pcollections.l, long, c4.m, com.duolingo.core.legacymodel.Direction, java.lang.String, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, org.pcollections.h, java.lang.String, org.pcollections.h, com.duolingo.shop.h, com.duolingo.user.GlobalAmbassadorStatus, java.lang.String, boolean, boolean, boolean, boolean, y7.f, java.lang.String, org.pcollections.l, long, int, java.lang.String, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.l, org.pcollections.l, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.h, org.pcollections.l, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, ia.j, boolean, org.pcollections.l, org.pcollections.l, org.pcollections.h, boolean, boolean, boolean, com.duolingo.user.StreakData, org.pcollections.l, java.lang.String, long, y4.r, java.lang.String, org.pcollections.l, com.duolingo.session.hi, boolean, com.duolingo.user.o, boolean, boolean, org.pcollections.l) failed to verify: void com.duolingo.user.q.<init>(com.duolingo.ads.AdsConfig, c4.k, com.duolingo.user.BetaStatus, java.lang.String, org.pcollections.l, org.pcollections.l, boolean, com.duolingo.shop.Outfit, org.pcollections.l, long, c4.m, com.duolingo.core.legacymodel.Direction, java.lang.String, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, org.pcollections.h, java.lang.String, org.pcollections.h, com.duolingo.shop.h, com.duolingo.user.GlobalAmbassadorStatus, java.lang.String, boolean, boolean, boolean, boolean, y7.f, java.lang.String, org.pcollections.l, long, int, java.lang.String, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.l, org.pcollections.l, java.lang.String, java.lang.String, org.pcollections.l, org.pcollections.h, org.pcollections.l, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, ia.j, boolean, org.pcollections.l, org.pcollections.l, org.pcollections.h, boolean, boolean, boolean, com.duolingo.user.StreakData, org.pcollections.l, java.lang.String, long, y4.r, java.lang.String, org.pcollections.l, com.duolingo.session.hi, boolean, com.duolingo.user.o, boolean, boolean, org.pcollections.l): [0x14] register v11 has type Undefined but expected Precise Reference: java.lang.String (declaration of 'com.duolingo.user.q' appears in /data/app/~~bvl59fr-yJPGNvh67taehQ==/com.duolingo-bfnCO2Wma8kKJsNNeVKGpQ==/base.apk!classes4.dex)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at o3.p0.I(Unknown Source:104)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at com.duolingo.core.repositories.e2.e(Unknown Source:12)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at com.duolingo.core.repositories.e2.c(Unknown Source:10)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at com.duolingo.core.repositories.e2$c.apply(Unknown Source:24)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at dl.z1$b.onNext(Unknown Source:27)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at dl.a1$c.d(Unknown Source:39)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at dl.a1$a.run(Unknown Source:17)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	at il.l.run(Unknown Source:21)
08-28 12:32:03.020 10709 10828 E AndroidRuntime: 	... 6 more
```


### Screenshots or videos

![photo_2023-08-28_14-08-11](https://github.com/ReVanced/revanced-patches/assets/39136587/ae41dfaa-abf3-45a4-8013-2816cde45bd1)

### Solution

n/a

### Additional context

n/a

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

--- a/src/main/kotlin/app/revanced/patches/duolingo/unlocksuper/UnlockDuolingoSuperPatch.kt
+++ b/src/main/kotlin/app/revanced/patches/duolingo/unlocksuper/UnlockDuolingoSuperPatch.kt
@@ -54,8 +54,8 @@ object UnlockDuolingoSuperPatch : BytecodePatch(
     }
 
     private fun MutableMethod.indexOfReference(reference: Reference) = getInstructions()
-        .filterIsInstance<BuilderInstruction22c>()
-        .filter { it.opcode == Opcode.IPUT_BOOLEAN }.indexOfFirst { it.reference == reference }.let {
+        .indexOfFirst { it is BuilderInstruction22c && it.opcode == Opcode.IPUT_BOOLEAN && it.reference == reference }
+        .let {
             if (it == -1) throw PatchException("Could not find index of instruction with supplied reference.")
             else it
         }

